lib/Sabre/OpenIdSabreAuthBackend.php

```diff
@@ -110,18 +110,23 @@ private function isDavAuthenticated($username) {
 	protected function validateBearerToken($bearerToken) {
 		if ($this->userSession->isLoggedIn() &&
 			$this->isDavAuthenticated($this->userSession->getUser()->getUID())) {
+			try {
 
-			// verify the bearer token
-			$tokenUser = $this->authModule->authToken($bearerToken);
-			if ($tokenUser === null) {
+				// verify the bearer token
+				$tokenUser = $this->authModule->authToken($bearerToken);
+				if ($tokenUser === null) {
+					return false;
+				}
+
+				// setup the user
+				$userId = $this->userSession->getUser()->getUID();
+				$this->setupFilesystem($userId);
+				$this->session->close();
+				return $this->principalPrefix . $userId;
+			} catch (\Exception $ex) {
+				$this->session->close();
```

Comment:
I would expect code that checks for expired token and then do some actions to refresh it. 
I don't understand what is going on here. some comments would be helpful.
I have an easy reproducer to test the 10min timeout, so whatever it does, I can easily veryfy, if it is successful afterwards :-)

Reply:
> I would expect code that checks for expired token and then do some actions to refresh it.

We as the reply party cannot refresh the token - this is upon the clients to take care.
(The server has no refresh token)